doc/sag/sag-HLD.md

```diff
@@ -0,0 +1,185 @@
+# SAG high level design for SONiC
+
+# Table of Contents
+- [SAG high level design for SONiC](#sag-high-level-design-for-sonic)
+- [Table of Contents](#table-of-contents)
+- [Revision History](#revision-history)
+- [Abbreviations](#abbreviations)
+- [Requirements Overview](#requirements-overview)
+- [Modules Design](#modules-design)
+  - [Config DB](#config-db)
+    - [SAG Global Table](#sag-global-table)
+    - [SAG Interface Table](#sag-interface-table)
+    - [Config DB Schemas](#config-db-schemas)
+  - [APP DB](#config-db)
+    - [Interface Table](#interface-table)
+    - [APP DB Schemas](#app-db-schemas)
+  - [Orchestration Agent](#orchestration-agent)
+    - [SagOrch](#sagorch)
+    - [SagMgrD](#sagmgrd)
+    - [IntfsOrch](#intfsorch)
+  - [SAI](#sai)
+- [Flows](#flows)  
+- [CLI](#cli)
+- [Test](#test)
+  - [Functional Test Cases](#functional-test-cases)
+
+# Revision History
+|  Rev  | Date       |   Author    | Change Description                |
+| :---: | :--------- | :---------: | --------------------------------- |
+|  0.1  | 08/13/2021 |  Jimi Chen  | Initial version                   |
+
+# Abbreviations
+| Abbreviations | Description            |
+|---------------|------------------------|
+| SAG           | Static Anycast Gateway |
+
+# Requirements Overview
+Anycast gateway feature for EVPN/VXLAN is a default gateway addressing mechanism that enables the use of the same gateway IP address across all the leaf switches that are part of a VXLAN network.  
+This ensures that every leaf switch can function as the default gateway for the workloads directly connected to it.  
+The feature failitates flexible workload placement, host mobility and optimal traffic forwarding across the VXLAN fabric.  
+
+This feature should be used together with VxLAN EVPN.  
+It should always to configure VxLAN to make SAG work.  
+SAG announces its IP and MAC only towards the host facing Ethernet ports but not towards the fabric (Overlay).  
+As a result, the IP/MAC for the SAG is only seen and reachable from the host facing side and never from the fabric (Overlay) facing side.
+
+The following is the requirements:  
+* Support one global virtual MAC address on SAG interface
+* Support multiple IPv4/IPv6 address on SAG interface
+* Support global IPv4/IPv6 enable knob for SAG address
+
+# Modules Design
+## Config DB
+### SAG Global Table
+```
+SAG_GLOBAL|IP 
+  "gwmac": {{mac_address}} 
+  "IPv4": {{enable}}
+  "IPv6": {{enable}}
+```
+### SAG Interface Table
+```
+SAG|{{vlan_intf_name}}|IPv4
+  "gwip@": {{ipv4_list}}
+SAG|{{vlan_intf_name}}|IPv6
+  "gwip@": {{ipv6_ilst}}
+```
```

Comment:
Please add config examples for these schemas.

Reply:
Try to use the below ip-version, i.e ipv4/ipv6.

typedef ip-version {
    type enumeration {
      enum unknown {
        value "0";
        description
         "An unknown or unspecified version of the Internet
          protocol.";
      }
      enum ipv4 {
        value "1";
        description
         "The IPv4 protocol as defined in RFC 791.";
      }
      enum ipv6 {
        value "2";
        description
         "The IPv6 protocol as defined in RFC 2460.";
      }
    }
    description
     "This value represents the version of the IP protocol.

      In the value set and its semantics, this type is equivalent
      to the InetVersion textual convention of the SMIv2.";
    reference
     "RFC  791: Internet Protocol
      RFC 2460: Internet Protocol, Version 6 (IPv6) Specification
      RFC 4001: Textual Conventions for Internet Network Addresses";
  }

Reply:
I think we should have it added to the vlan interface schema instead of creating new block.

Reply:
Refined the DB schema design, the IP address will be the same as original VLAN interface configuration.  
Introduce a knob to enable/disable SAG on the VLAN interface, this can switch the MAC address between system CPU MAC address and virutal SAG MAC address.